diff --git a/sample_apps/mfg/uart_wifi_bridge/src/uart_wifi_bridge.h b/sample_apps/mfg/uart_wifi_bridge/src/uart_wifi_bridge.h
index 88a3d7d..0576c20 100644
--- a/sample_apps/mfg/uart_wifi_bridge/src/uart_wifi_bridge.h
+++ b/sample_apps/mfg/uart_wifi_bridge/src/uart_wifi_bridge.h
@@ -28,7 +28,7 @@
 
 #include <compat_attribute.h>
 
-#define BAUD_RATE 1500000
+#define BAUD_RATE 115200
 
 /* Configure BUF_LEN : length of the buffer used to send wifi packets */
 #define BUF_LEN 1024
